themes/apiarybasestyles/source/css/buttons.styl

```diff
@@ -10,3 +10,10 @@
     background-color #264653
     &:hover
       background-color #2A9D8F
+
+.loadingIndicator
+  $loadingIndicator()
+  &--big
+    $loadingIndicator(size: 12px)
+  &--blue
+    $loadingIndicator(color: $primaryColor)
```

Comment:
That might be kinda painful when you want base styles to also take care of hover states or even animations (in a case of loader indicator).

And in a case of a primary/secondary button - primary button would be 1 `@extend` and secondary would be 1 `@extend` + manually added CSS with colors, hover…?